src/AppInstallerCommonCore/Public/AppInstallerLogging.h

```diff
@@ -140,6 +140,37 @@ namespace AppInstaller::Logging
 
     // Calls the various stream format functions to produce an 8 character hexadecimal output.
     std::ostream& SetHRFormat(std::ostream& out);
+
+    // This type allows us to override the default behavior of output operators for logging.
+    struct LoggingStream
+    {
+        // Force use of the UTF-8 string from a file path.
+        // This should not be necessary when we move to C++20 and convert to using u8string.
+        friend AppInstaller::Logging::LoggingStream& operator<<(AppInstaller::Logging::LoggingStream& out, std::filesystem::path& path)

```

Comment:
>std::filesystem::path& path [](start = 107, length = 27)

Is this override with non const variant used?

Reply:
Yes; when doing overload resolution, only the exact match will be considered a better candidate than the template.  So a non-const local will not match the const& function.  Unfortunately I'm not aware of a better way to force the compiler to "do everything as you would have except in this one case".  I suppose it might be possible to add some dummy parameters to the template to make it a worse candidate function.